viewing_party/party.py

```diff
@@ -1,23 +1,201 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    new_movie = {}
+    
+    if not title or not genre or not rating:
+        return None
+    
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"] += [movie]
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"] += [movie]
+
+    return user_data
+
+def watch_movie(user_data, title):
+
+    watchlist = user_data["watchlist"]
+
+    for i in range(len(watchlist)):
+        if title == watchlist[i]["title"]:
+            user_data["watched"] += [watchlist[i]]
+            watchlist.pop(i)

```

Comment:
Again, this is another creative solution! Making use of the indexes in your range loop to pass the specified index to your `pop` method. A more common solution you might see is this: 
```python
def watch_movie(user_data, title):
    for movie in user_data["watchlist"]:
        if movie["title"] == title:
            user_data["watchlist"].remove(movie)
            user_data["watched"].append(movie)

            return user_data

    return user_data
```